chemprop/data/splitting.py

```diff
@@ -25,8 +26,8 @@ class SplitType(EnumMapping):
     KMEANS = auto()
 
 
-def split_data(
-    datapoints: Sequence[MoleculeDatapoint],
+def make_split_indices(
+    mols: Sequence[Chem.Mol],
```

Comment:
A prior limitation of split_data/make_split_indices was that it only took `MoleculeDatapoint`. This means the function couldn't be used to split by reactants of a reaction dataset. The function only used the molecules to get the RDkit Mol objects `[d.mol for d in datapoints]` for structure based splits. I made the function more flexible by having it accept this list of Mol objects directly. In the CLI, if the datapoints to use for splitting are molecules, this list of Mol objects is made from the data.mol attributes. If the datapoints are reactions, then the data.rct reactant Mols are used. It seems to be the case that people will more often split by reactants instead of products, so I didn't feel the option to split by products was worth a new CLI flag.